### PR TITLE
Miawong/sc 75115/improve layout for long strings

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it: This PR improves the ability for admin console to handle long strings in app name, license, and config titles. 
***********BEFORE***********
Screenshot 2023-05-01 at 07 38 52

Screenshot 2023-05-02 at 4 25 24 PM
Screenshot 2023-05-01 at 07 38 37

**********AFTER**********
Screenshot 2023-05-02 at 4 30 11 PM

Screenshot 2023-05-03 at 10 14 47 AM
Screenshot 2023-05-02 at 4 32 34 PM

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/75115/improve-layout-for-long-strings

#### Does this PR introduce a user-facing change?

```release-note
Enhances formatting to accommodate lengthy strings.
```

### CODE REVIEW

File: web/src/components/config_render/ConfigItemTitle.jsx [hunk 1]
```diff
@@ -24,10 +24,10 @@ export default class ConfigItemTitle extends React.Component {
     return (
       <h4
         className="card-item-title field-section-sub-header"
-        style={title ? { marginBottom: -18 } : {}}
+        style={title?.length < 101 ? { marginBottom: -18 } : {}}
```
Comment (miaawong) [inline web/src/components/config_render/ConfigItemTitle.jsx:27]: No marginBottom if we have title length of more than 101. The "required/recommended" tags will overlap with title if theres a negative margin.

File: web/src/features/AppConfig/components/AppConfig.tsx [hunk 2]
```diff
@@ -752,66 +752,67 @@ class AppConfig extends Component<Props, State> {
...
-                  className={`side-nav-group ${
-                    this.state.activeGroups.includes(group.name) ||
-                    group.hasError
-                      ? "group-open"
-                      : ""
-                  }`}
-                  id={`config-group-nav-${group.name}`}
-                >
+        <div className="tw-flex tw-flex-col tw-mx-48">
+          {fromLicenseFlow && app && (
```
Comment (miaawong) [inline web/src/features/AppConfig/components/AppConfig.tsx:756]: added some div - flex to fix the alignment of the app name and the config section.

File: web/src/features/Auth/components/SecureAdminConsole.tsx [hunk 3]
```diff
@@ -196,7 +196,7 @@ class SecureAdminConsole extends React.Component<Props, State> {
       this.loginText.current.innerHTML,
       this.loginText.current.clientWidth,
       "32px",
-      null
+      "10px"
```
Comment (miaawong) [inline web/src/features/Auth/components/SecureAdminConsole.tsx:199]: there's a resize function in utils that will resize the app name, but this doesn't occur every time. Need to look into it more.

File: web/src/utilities/utilities.js [hunk 4]
```diff
@@ -155,8 +155,12 @@ export function dynamicallyResizeText(
   }
 
   while (resizerElm.getBoundingClientRect().width > maxWidth) {
-    size = parseInt(resizerElm.style.fontSize, 10);
-    resizerElm.style.fontSize = `${size - 1}px`;
+    if (resizerElm.style.fontSize > 10) {
+      size = parseInt(resizerElm.style.fontSize, 10);
+      resizerElm.style.fontSize = `${size - 1}px`;
+    } else {
+      return `${minFontSize}`;
```
Comment (miaawong) [inline web/src/utilities/utilities.js:162]: if we have really long strings, this function can make font sizes too small (2px). I am limiting it to minimum of 10px for now.

File: web/src/scss/utilities/utilities.scss [hunk 5]
```diff
@@ -341,3 +341,8 @@
 .nowrap {
   white-space: nowrap;
 }
+
+.break-word {
+  word-break: normal;
+  overflow-wrap: anywhere;
```
Comment (miaawong) [inline web/src/scss/utilities/utilities.scss:347]: breaks anywhere, when the string overflows the container

File: web/src/features/Auth/components/SecureAdminConsole.tsx (hunk 3 above)
Comment (rianfowler) [reply web/src/features/Auth/components/SecureAdminConsole.tsx:199]: interesting

**Review by rianfowler — DISMISSED** [review]
nice work!

**Review by rianfowler — APPROVED** [review]

**Pull request merged**